Sources/PitchSpeller/Data Structures/Pairable.swift

```diff
@@ -0,0 +1,20 @@
+//
+//  Pairable.swift
+//  PitchSpeller
+//
+//  Created by Benjamin Wetherfield on 6/30/18.
+//
+
+protocol Pairable {
+    associatedtype A
+    associatedtype B
+    var a: A { get }
+    var b: B { get }
+    var tuple: (A,B) { get }
+    init(_ pair: (A,B))
```

Comment:
Is it critical to demand both of these initializers? If we can get away with it, things may be a bit more agile down the line if only required the `init(_ a: A, _ b: B)` is required, if that.

Reply:
I have generally found that it is nice to only require things which are used within default implementations in protocol extensions. I have over-prescribed before, and it became a bit of a burden downstream.

Reply:
Cool. I may keep in `init(_ a: A, _b: A)` for the default definition of `.swapped`...?